upf/grpc/upf.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+message Fluent {
+    string name = 1;
+    string valueType = 2;
+    repeated string signature = 3;
+}
+
+message Object {
+    string name = 1;
+    string type = 2; 
+}
+
+message Expression {
+    int64 type = 1;
+    repeated Expression args = 2;
+    Payload payload = 3;
+}
+
+message Assignment {
+    Expression x = 1;
+    Expression v = 2;
+}
+
+message Payload {
+    string type = 1;
+    string value = 2;
+}
+
+message Action {
+    string name = 1;
+    repeated string parameters = 2;
```

Comment:
Instead of two fields `parameters` and `parameterTypes`, shouldn't this be a single sequence 
`repeated Parameter parameters`, where `Parameter contains the `name` and the `type`. Note that `Parameter` would be the same as `Object` but with a different semantic. It might evolve into something different in the future.

This enforces the fact that each parameter is given a type.